lua/claudecode/tools/get_diagnostics.lua

```diff
@@ -17,29 +32,78 @@ local function handler(_params) -- Prefix unused params with underscore
     })
   end
 
-  local all_diagnostics = vim.diagnostic.get(0) -- Get for all buffers
+  local log_module_ok, log = pcall(require, "claudecode.logger")
+  if not log_module_ok then
+    return {
+      code = -32603, -- Internal error
+      message = "Internal error",
+      data = "Failed to load logger module",
+    }
+  end
+
+  log.debug("getDiagnostics handler called with params: " .. vim.inspect(params))
+
+  -- Extract the uri parameter
+  local diagnostics
+
+  if not params.uri then
+    -- Get diagnostics for all buffers
+    log.debug("Getting diagnostics for all open buffers")
+    diagnostics = vim.diagnostic.get()
+  else
+    -- Remove file:// prefix if present
+    local uri = params.uri
+    local filepath = uri
+    if uri:sub(1, 7) == "file://" then
+      filepath = uri:sub(8) -- Remove "file://" prefix
+    end
+
+    -- Get buffer number for the specific file
+    local bufnr = vim.fn.bufnr(filepath)
+    if bufnr == -1 then
+      -- File is not open in any buffer, throw an error
+      log.debug("File buffer must be open to get diagnostics: " .. filepath)
+      error({
+        code = -32001,
+        message = "File not open in buffer",
+        data = "File must be open in Neovim to retrieve diagnostics: " .. filepath,
+      })
+    else
+      -- Get diagnostics for the specific buffer
+      log.debug("Getting diagnostics for bufnr: " .. bufnr)
+      diagnostics = vim.diagnostic.get(bufnr)
+    end
+  end
 
   local formatted_diagnostics = {}
-  for _, diagnostic in ipairs(all_diagnostics) do
+  for _, diagnostic in ipairs(diagnostics) do
     local file_path = vim.api.nvim_buf_get_name(diagnostic.bufnr)
     -- Ensure we only include diagnostics with valid file paths
     if file_path and file_path ~= "" then
       table.insert(formatted_diagnostics, {
-        file = file_path,
-        line = diagnostic.lnum, -- 0-indexed from vim.diagnostic.get
-        character = diagnostic.col, -- 0-indexed from vim.diagnostic.get
-        severity = diagnostic.severity, -- e.g., vim.diagnostic.severity.ERROR
-        message = diagnostic.message,
-        source = diagnostic.source,
+        type = "text",
+        -- json encode this
+        text = vim.fn.json_encode({
+          -- Use the file path and diagnostic information
+          filePath = file_path,
+          -- Convert line and column to 1-indexed
+          line = diagnostic.lnum + 1,
+          character = diagnostic.col + 1,
+          severity = diagnostic.severity, -- e.g., vim.diagnostic.severity.ERROR
+          message = diagnostic.message,
+          source = diagnostic.source,
+        }),
```

Comment:
Prefer `vim.json.encode` over `vim.fn.json_encode` for speed and no Vim-Lua round-trip.